antarest/study/business/link_management.py

```diff
@@ -14,30 +14,152 @@
 
 from pydantic import BaseModel
 
-from antarest.core.exceptions import ConfigFileNotFound
+from antarest.core.exceptions import ConfigFileNotFound, InvalidFieldForVersionError
 from antarest.core.model import JSON
 from antarest.study.business.all_optional_meta import all_optional_model, camel_case_model
 from antarest.study.business.utils import execute_or_add_commands
 from antarest.study.model import RawStudy
-from antarest.study.storage.rawstudy.model.filesystem.config.links import LinkProperties
+from antarest.study.storage.rawstudy.model.filesystem.config.links import (
+    AssetType,
+    LinkProperties,
+    LinkStyle,
+    TransmissionCapacity,
+)
 from antarest.study.storage.storage_service import StudyStorageService
+from antarest.study.storage.variantstudy.model.command.common import FilteringOptions
 from antarest.study.storage.variantstudy.model.command.create_link import CreateLink
 from antarest.study.storage.variantstudy.model.command.remove_link import RemoveLink
 from antarest.study.storage.variantstudy.model.command.update_config import UpdateConfig
 
 _ALL_LINKS_PATH = "input/links"
+DEFAULT_COLOR = "112"
 
 
-class LinkUIDTO(BaseModel):
-    color: str
-    width: float
-    style: str
-
-
-class LinkInfoDTO(BaseModel):
+class LinkInfoDTOBase(BaseModel):
     area1: str
     area2: str
-    ui: t.Optional[LinkUIDTO] = None
+    hurdles_cost: t.Optional[bool] = False
+    loop_flow: t.Optional[bool] = False
+    use_phase_shifter: t.Optional[bool] = False
+    transmission_capacities: t.Optional[str] = TransmissionCapacity.ENABLED
+    asset_type: t.Optional[str] = AssetType.AC
+    display_comments: t.Optional[bool] = True
+    colorr: t.Optional[str] = DEFAULT_COLOR
+    colorb: t.Optional[str] = DEFAULT_COLOR
+    colorg: t.Optional[str] = DEFAULT_COLOR
+    link_width: t.Optional[float] = 1
+    link_style: t.Optional[str] = LinkStyle.PLAIN
+
+
+class LinkInfoDTO820(LinkInfoDTOBase):
+    filter_synthesis: t.Optional[str] = None
+    filter_year_by_year: t.Optional[str] = None
+
+
+LinkInfoDTOType = t.Union[LinkInfoDTO820, LinkInfoDTOBase]
+
+
+class LinkInfoFactory:
+    @staticmethod
+    def create_link_info(version: int, **kwargs) -> LinkInfoDTOType:
```

Comment:
type `**kwargs: t.Any` it will solve mypy issue (same for other warnings in this file)